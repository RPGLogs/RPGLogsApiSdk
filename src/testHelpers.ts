import { buildSdk } from './main';
import btoa from 'btoa';

export function buildTestSdk(): ReturnType<typeof buildSdk> {
  return buildSdk(<string>process.env.TEST_ACCESS_TOKEN);
}

export async function getAccessToken(
  clientId: string,
  clientSecret: string,
  game = 'warcraft'
): Promise<string> {
  if (game === 'warcraft') game = 'www.warcraft';

  const authHeader = 'Basic ' + btoa(clientId + ':' + clientSecret);

  const response = await fetch(`https://${game}logs.com/oauth/token`, {
    method: 'POST',
    headers: {
      Authorization: authHeader,
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: 'grant_type=client_credentials',
  });

  const json = await response.json();

  if (response.status === 200) {
    return json.access_token;
  } else {
    throw new Error('Response was not OK: ' + JSON.stringify(json ?? {}));
  }
}
