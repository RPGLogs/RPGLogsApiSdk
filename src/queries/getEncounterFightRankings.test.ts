import { buildTestSdk } from '../testHelpers';
import { FightRankingMetricType } from '../sdk';

test('returns data', async () => {
  const response = await buildTestSdk().getEncounterFightRankings({
    encounterId: 2333,
    metric: FightRankingMetricType.Speed,
    difficulty: 5,
  });

  expect(response?.worldData?.encounter).toMatchSnapshot();
});
