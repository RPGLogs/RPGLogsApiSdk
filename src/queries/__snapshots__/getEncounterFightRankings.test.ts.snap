// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns data 1`] = `
Object {
  "fightRankings": Object {
    "count": 50,
    "hasMorePages": true,
    "page": 1,
    "rankings": Array [
      Object {
        "bracketData": 478.85,
        "damageTaken": 24805168,
        "deaths": 0,
        "duration": 75052,
        "guild": Object {
          "faction": 0,
          "id": 6956,
          "name": "Honestly",
        },
        "healers": 0,
        "melee": 3,
        "ranged": 16,
        "report": Object {
          "code": "nhjg1p4kBMr39GYz",
          "fightID": 28,
          "startTime": 1599383830672,
        },
        "server": Object {
          "id": 99,
          "name": "Frostmourne",
          "region": "US",
        },
        "startTime": 1599388705134,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.45,
        "damageTaken": 39292516,
        "deaths": 7,
        "duration": 79259,
        "guild": Object {
          "faction": 1,
          "id": 369203,
          "name": "Reaction",
        },
        "healers": 0,
        "melee": 7,
        "ranged": 12,
        "report": Object {
          "code": "ANw6Fynq8W3P1Bkg",
          "fightID": 15,
          "startTime": 1602092886405,
        },
        "server": Object {
          "id": 304,
          "name": "Kazzak",
          "region": "EU",
        },
        "startTime": 1602097935818,
        "tanks": 1,
      },
      Object {
        "bracketData": 480.25,
        "damageTaken": 35081151,
        "deaths": 2,
        "duration": 80280,
        "guild": Object {
          "faction": 0,
          "id": 395,
          "name": "Vindicatum",
        },
        "healers": 0,
        "melee": 7,
        "ranged": 12,
        "report": Object {
          "code": "XNyTkmDGWQaxjVH7",
          "fightID": 57,
          "startTime": 1601251210248,
        },
        "server": Object {
          "id": 122,
          "name": "Icecrown",
          "region": "US",
        },
        "startTime": 1601265597400,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.15,
        "damageTaken": 41660316,
        "deaths": 6,
        "duration": 81192,
        "guild": Object {
          "faction": 1,
          "id": 12337,
          "name": "Fused",
        },
        "healers": 0,
        "melee": 8,
        "ranged": 11,
        "report": Object {
          "code": "MaqL183nhkw4gTAW",
          "fightID": 50,
          "startTime": 1595981346954,
        },
        "server": Object {
          "id": 123,
          "name": "Illidan",
          "region": "US",
        },
        "startTime": 1595992634128,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.6,
        "damageTaken": 40512978,
        "deaths": 12,
        "duration": 82696,
        "guild": Object {
          "faction": 1,
          "id": 100393,
          "name": "xD",
        },
        "healers": 0,
        "melee": 7,
        "ranged": 12,
        "report": Object {
          "code": "tgcC7qx94bZGQBdz",
          "fightID": 24,
          "startTime": 1600908513332,
        },
        "server": Object {
          "id": 18,
          "name": "Area 52",
          "region": "US",
        },
        "startTime": 1600912630524,
        "tanks": 1,
      },
      Object {
        "bracketData": 478.55,
        "damageTaken": 32061090,
        "deaths": 4,
        "duration": 83670,
        "guild": Object {
          "faction": 1,
          "id": 507188,
          "name": "poptart corndoG",
        },
        "healers": 0,
        "melee": 5,
        "ranged": 14,
        "report": Object {
          "code": "kLPfV8d6xvcDJWb4",
          "fightID": 33,
          "startTime": 1595989753024,
        },
        "server": Object {
          "id": 222,
          "name": "Tichondrius",
          "region": "US",
        },
        "startTime": 1595996370298,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.4,
        "damageTaken": 39267370,
        "deaths": 7,
        "duration": 84531,
        "guild": Object {
          "faction": 1,
          "id": 350026,
          "name": "Northern Sky",
        },
        "healers": 0,
        "melee": 4,
        "ranged": 15,
        "report": Object {
          "code": "JKgMDzC3dtHa4vRp",
          "fightID": 3,
          "startTime": 1594315918743,
        },
        "server": Object {
          "id": 406,
          "name": "Blackhand",
          "region": "EU",
        },
        "startTime": 1594316480681,
        "tanks": 1,
      },
      Object {
        "bracketData": 478.8,
        "damageTaken": 39916866,
        "deaths": 7,
        "duration": 84748,
        "guild": Object {
          "faction": 0,
          "id": 230717,
          "name": "Elicit",
        },
        "healers": 0,
        "melee": 5,
        "ranged": 14,
        "report": Object {
          "code": "yZTGgNmD2K1LQaAM",
          "fightID": 25,
          "startTime": 1599643158795,
        },
        "server": Object {
          "id": 99,
          "name": "Frostmourne",
          "region": "US",
        },
        "startTime": 1599647126127,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.4,
        "damageTaken": 46609457,
        "deaths": 5,
        "duration": 86571,
        "guild": Object {
          "faction": 1,
          "id": 381999,
          "name": "Virtual",
        },
        "healers": 0,
        "melee": 7,
        "ranged": 11,
        "report": Object {
          "code": "CHWygGNMXVQx6z9p",
          "fightID": 25,
          "startTime": 1596042456157,
        },
        "server": Object {
          "id": 350,
          "name": "Twisting Nether",
          "region": "EU",
        },
        "startTime": 1596046022034,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.35,
        "damageTaken": 48402992,
        "deaths": 0,
        "duration": 88762,
        "guild": Object {
          "faction": 1,
          "id": 10210,
          "name": "Colour",
        },
        "healers": 2,
        "melee": 4,
        "ranged": 12,
        "report": Object {
          "code": "rRDpj62cP9KHtdz4",
          "fightID": 8,
          "startTime": 1596131696761,
        },
        "server": Object {
          "id": 335,
          "name": "Stormreaver",
          "region": "EU",
        },
        "startTime": 1596132271526,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.6,
        "damageTaken": 49251366,
        "deaths": 2,
        "duration": 90661,
        "guild": Object {
          "faction": 1,
          "id": 461134,
          "name": "Райдо",
        },
        "healers": 1,
        "melee": 6,
        "ranged": 12,
        "report": Object {
          "code": "VNp82PR7Gvyc439X",
          "fightID": 20,
          "startTime": 1595438120565,
        },
        "server": Object {
          "id": 495,
          "name": "Ревущий фьорд",
          "region": "EU",
        },
        "startTime": 1595444538236,
        "tanks": 1,
      },
      Object {
        "bracketData": 478.65,
        "damageTaken": 54132102,
        "deaths": 1,
        "duration": 91570,
        "guild": Object {
          "faction": 1,
          "id": 361199,
          "name": "Casual Raiders",
        },
        "healers": 1,
        "melee": 6,
        "ranged": 12,
        "report": Object {
          "code": "wpTWNgaHZB42Lvb7",
          "fightID": 5,
          "startTime": 1594057995089,
        },
        "server": Object {
          "id": 340,
          "name": "Tarren Mill",
          "region": "EU",
        },
        "startTime": 1594060328603,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.55,
        "damageTaken": 45802004,
        "deaths": 1,
        "duration": 93932,
        "guild": Object {
          "faction": 1,
          "id": 19537,
          "name": "Spike Flail",
        },
        "healers": 0,
        "melee": 6,
        "ranged": 12,
        "report": Object {
          "code": "HF8y6dZLpVkvafP7",
          "fightID": 9,
          "startTime": 1594943726488,
        },
        "server": Object {
          "id": 150,
          "name": "Mal'Ganis",
          "region": "US",
        },
        "startTime": 1594950103099,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.1,
        "damageTaken": 49941538,
        "deaths": 1,
        "duration": 94913,
        "guild": Object {
          "faction": 0,
          "id": 87928,
          "name": "Инвинсибл",
        },
        "healers": 1,
        "melee": 7,
        "ranged": 11,
        "report": Object {
          "code": "ZTLtQ3vAx8PrdC2R",
          "fightID": 36,
          "startTime": 1600884251175,
        },
        "server": Object {
          "id": 488,
          "name": "Гордунни",
          "region": "EU",
        },
        "startTime": 1600892606422,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.75,
        "damageTaken": 64320473,
        "deaths": 7,
        "duration": 97166,
        "guild": Object {
          "faction": 1,
          "id": 492201,
          "name": "Sanity",
        },
        "healers": 1,
        "melee": 5,
        "ranged": 12,
        "report": Object {
          "code": "ZCgQJ1PLp9HG3vat",
          "fightID": 24,
          "startTime": 1597859595003,
        },
        "server": Object {
          "id": 283,
          "name": "Draenor",
          "region": "EU",
        },
        "startTime": 1597865114149,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.15,
        "damageTaken": 43694885,
        "deaths": 4,
        "duration": 99000,
        "guild": Object {
          "faction": 0,
          "id": 473786,
          "name": "ad hoc",
        },
        "healers": 0,
        "melee": 8,
        "ranged": 10,
        "report": Object {
          "code": "pTRtmQJgHdWLGBCq",
          "fightID": 35,
          "startTime": 1599695977604,
        },
        "server": Object {
          "id": 206,
          "name": "Stormrage",
          "region": "US",
        },
        "startTime": 1599701610136,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.75,
        "damageTaken": 50121884,
        "deaths": 8,
        "duration": 99295,
        "guild": Object {
          "faction": 0,
          "id": 448946,
          "name": "Consequence",
        },
        "healers": 0,
        "melee": 5,
        "ranged": 13,
        "report": Object {
          "code": "Y4bT1Fv7HZR36Ajz",
          "fightID": 30,
          "startTime": 1597799310848,
        },
        "server": Object {
          "id": 206,
          "name": "Stormrage",
          "region": "US",
        },
        "startTime": 1597803994006,
        "tanks": 2,
      },
      Object {
        "bracketData": 478.25,
        "damageTaken": 60939426,
        "deaths": 2,
        "duration": 99798,
        "guild": Object {
          "faction": 1,
          "id": 372899,
          "name": "Counting Globals",
        },
        "healers": 1,
        "melee": 5,
        "ranged": 13,
        "report": Object {
          "code": "XB2AmpYq7djT4FnN",
          "fightID": 2,
          "startTime": 1592503155689,
        },
        "server": Object {
          "id": 304,
          "name": "Kazzak",
          "region": "EU",
        },
        "startTime": 1592503396194,
        "tanks": 1,
      },
      Object {
        "bracketData": 479.1,
        "damageTaken": 63625982,
        "deaths": 4,
        "duration": 99847,
        "guild": Object {
          "faction": 1,
          "id": 529369,
          "name": "Экзекьют",
        },
        "healers": 1,
        "melee": 6,
        "ranged": 12,
        "report": Object {
          "code": "YhG9NJR8zf1kZm2v",
          "fightID": 44,
          "startTime": 1602090063583,
        },
        "server": Object {
          "id": 495,
          "name": "Ревущий фьорд",
          "region": "EU",
        },
        "startTime": 1602103258015,
        "tanks": 1,
      },
      Object {
        "bracketData": 478.9,
        "damageTaken": 40516110,
        "deaths": 4,
        "duration": 100821,
        "guild": Object {
          "faction": 0,
          "id": 514858,
          "name": "Лё Резистанс",
        },
        "healers": 0,
        "melee": 6,
        "ranged": 13,
        "report": Object {
          "code": "LfKchndNpXGaMWxz",
          "fightID": 24,
          "startTime": 1600276432766,
        },
        "server": Object {
          "id": 488,
          "name": "Гордунни",
          "region": "EU",
        },
        "startTime": 1600281518357,
        "tanks": 1,
      },
      Object {
        "bracketData": 480.25,
        "damageTaken": 54408697,
        "deaths": 1,
        "duration": 101186,
        "guild": Object {
          "faction": 1,
          "id": 559813,
          "name": "燃烧之刃-黑手军团",
        },
        "healers": 1,
        "melee": 3,
        "ranged": 15,
        "report": Object {
          "code": "L4mdfQ2zVb3gXh76",
          "fightID": 20,
          "startTime": 1602417350263,
        },
        "server": Object {
          "id": 750,
          "name": "燃烧之刃",
          "region": "CN",
        },
        "startTime": 1602420179174,
        "tanks": 1,
      },
      Object {
        "bracketData": 480.35,
        "damageTaken": 60706790,
        "deaths": 4,
        "duration": 102706,
        "guild": Object {
          "faction": 0,
          "id": 7731,
          "name": "永恒理想",
        },
        "healers": 1,
        "melee": 4,
        "ranged": 13,
        "report": Object {
          "code": "MPx61KRYLbfy3mD7",
          "fightID": 19,
          "startTime": 1599739719949,
        },
        "server": Object {
          "id": 564,
          "name": "主宰之剑",
          "region": "CN",
        },
        "startTime": 1599742644693,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.45,
        "damageTaken": 46928093,
        "deaths": 1,
        "duration": 103368,
        "guild": Object {
          "faction": 1,
          "id": 324679,
          "name": "it's my life",
        },
        "healers": 1,
        "melee": 3,
        "ranged": 14,
        "report": Object {
          "code": "Mkj9YtRAvhF2WmDw",
          "fightID": 4,
          "startTime": 1599135985250,
        },
        "server": Object {
          "id": 901,
          "name": "黑铁",
          "region": "CN",
        },
        "startTime": 1599138907272,
        "tanks": 2,
      },
      Object {
        "bracketData": 478.85,
        "damageTaken": 50284567,
        "deaths": 0,
        "duration": 103985,
        "guild": Object {
          "faction": 1,
          "id": 22317,
          "name": "Hyped",
        },
        "healers": 3,
        "melee": 6,
        "ranged": 9,
        "report": Object {
          "code": "z3t9LvphVaFGkRnd",
          "fightID": 39,
          "startTime": 1598551260907,
        },
        "server": Object {
          "id": 340,
          "name": "Tarren Mill",
          "region": "EU",
        },
        "startTime": 1598559685411,
        "tanks": 2,
      },
      Object {
        "bracketData": 478.5,
        "damageTaken": 59942454,
        "deaths": 3,
        "duration": 105381,
        "guild": Object {
          "faction": 1,
          "id": 359204,
          "name": "Big Damage",
        },
        "healers": 1,
        "melee": 8,
        "ranged": 10,
        "report": Object {
          "code": "gzNwqybnvc6FA47m",
          "fightID": 34,
          "startTime": 1600560627912,
        },
        "server": Object {
          "id": 18,
          "name": "Area 52",
          "region": "US",
        },
        "startTime": 1600565446976,
        "tanks": 1,
      },
      Object {
        "bracketData": 480,
        "damageTaken": 63186548,
        "deaths": 1,
        "duration": 106139,
        "guild": Object {
          "faction": 1,
          "id": 20616,
          "name": "Twisted Vortex",
        },
        "healers": 2,
        "melee": 5,
        "ranged": 11,
        "report": Object {
          "code": "aLRKmgwXc214hFjW",
          "fightID": 34,
          "startTime": 1601431189931,
        },
        "server": Object {
          "id": 25,
          "name": "Azralon",
          "region": "US",
        },
        "startTime": 1601436081085,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.05,
        "damageTaken": 61139942,
        "deaths": 0,
        "duration": 107619,
        "guild": Object {
          "faction": 1,
          "id": 238024,
          "name": "Pescorus",
        },
        "healers": 1,
        "melee": 8,
        "ranged": 9,
        "report": Object {
          "code": "KdbN8Rfjac3Vr9Wh",
          "fightID": 26,
          "startTime": 1602179918021,
        },
        "server": Object {
          "id": 304,
          "name": "Kazzak",
          "region": "EU",
        },
        "startTime": 1602186428398,
        "tanks": 2,
      },
      Object {
        "bracketData": 478,
        "damageTaken": 53756873,
        "deaths": 0,
        "duration": 107713,
        "guild": Object {
          "faction": 1,
          "id": 511071,
          "name": "endeavør",
        },
        "healers": 1,
        "melee": 4,
        "ranged": 13,
        "report": Object {
          "code": "Fb8Jk2p4nvXh1dmx",
          "fightID": 48,
          "startTime": 1591813665454,
        },
        "server": Object {
          "id": 370,
          "name": "Hyjal",
          "region": "EU",
        },
        "startTime": 1591823532930,
        "tanks": 2,
      },
      Object {
        "bracketData": 478.95,
        "damageTaken": 59249641,
        "deaths": 2,
        "duration": 108386,
        "guild": Object {
          "faction": 1,
          "id": 1031,
          "name": "gn",
        },
        "healers": 1,
        "melee": 5,
        "ranged": 12,
        "report": Object {
          "code": "8V4XpRLfTjqxnHNG",
          "fightID": 18,
          "startTime": 1594771326872,
        },
        "server": Object {
          "id": 150,
          "name": "Mal'Ganis",
          "region": "US",
        },
        "startTime": 1594773216181,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.75,
        "damageTaken": 56798082,
        "deaths": 3,
        "duration": 109160,
        "guild": Object {
          "faction": 1,
          "id": 2046,
          "name": "Wiping As Intended",
        },
        "healers": 2,
        "melee": 4,
        "ranged": 12,
        "report": Object {
          "code": "fHYWtN3d6XQCbKTF",
          "fightID": 11,
          "startTime": 1597860235994,
        },
        "server": Object {
          "id": 283,
          "name": "Draenor",
          "region": "EU",
        },
        "startTime": 1597864184506,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.6,
        "damageTaken": 59868361,
        "deaths": 2,
        "duration": 111034,
        "guild": Object {
          "faction": 1,
          "id": 95321,
          "name": "Dinosaur Cowboys",
        },
        "healers": 1,
        "melee": 7,
        "ranged": 10,
        "report": Object {
          "code": "KZ8PHhcdaQRr3VqC",
          "fightID": 22,
          "startTime": 1600819458992,
        },
        "server": Object {
          "id": 18,
          "name": "Area 52",
          "region": "US",
        },
        "startTime": 1600824618372,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.7,
        "damageTaken": 74995122,
        "deaths": 1,
        "duration": 111078,
        "guild": Object {
          "faction": 1,
          "id": 54728,
          "name": "Security",
        },
        "healers": 2,
        "melee": 8,
        "ranged": 8,
        "report": Object {
          "code": "vkw2bcTB9A6gDpYR",
          "fightID": 27,
          "startTime": 1599068032177,
        },
        "server": Object {
          "id": 340,
          "name": "Tarren Mill",
          "region": "EU",
        },
        "startTime": 1599071885317,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.15,
        "damageTaken": 67279444,
        "deaths": 5,
        "duration": 111359,
        "guild": Object {
          "faction": 0,
          "id": 346132,
          "name": "Versatile",
        },
        "healers": 1,
        "melee": 5,
        "ranged": 12,
        "report": Object {
          "code": "dWvyzrMp6VJRYjct",
          "fightID": 11,
          "startTime": 1597861735779,
        },
        "server": Object {
          "id": 329,
          "name": "Silvermoon",
          "region": "EU",
        },
        "startTime": 1597865299440,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.3,
        "damageTaken": 61226004,
        "deaths": 2,
        "duration": 112244,
        "guild": Object {
          "faction": 1,
          "id": 494425,
          "name": "SliverHand_9.0",
        },
        "healers": 2,
        "melee": 4,
        "ranged": 12,
        "report": Object {
          "code": "t6c7nyXGARrHDFMj",
          "fightID": 37,
          "startTime": 1599996676505,
        },
        "server": Object {
          "id": 770,
          "name": "白银之手",
          "region": "CN",
        },
        "startTime": 1600001372148,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.95,
        "damageTaken": 54435935,
        "deaths": 0,
        "duration": 112521,
        "guild": Object {
          "faction": 1,
          "id": 517468,
          "name": "Ookers",
        },
        "healers": 2,
        "melee": 3,
        "ranged": 13,
        "report": Object {
          "code": "TnJkQ8m9Lg3aCX6B",
          "fightID": 27,
          "startTime": 1600215549400,
        },
        "server": Object {
          "id": 123,
          "name": "Illidan",
          "region": "US",
        },
        "startTime": 1600223364149,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.6,
        "damageTaken": 62122766,
        "deaths": 5,
        "duration": 112999,
        "guild": Object {
          "faction": 1,
          "id": 493088,
          "name": "Imagine Progress",
        },
        "healers": 1,
        "melee": 4,
        "ranged": 13,
        "report": Object {
          "code": "kF9g72WxLQtdfTYZ",
          "fightID": 9,
          "startTime": 1597802457301,
        },
        "server": Object {
          "id": 150,
          "name": "Mal'Ganis",
          "region": "US",
        },
        "startTime": 1597807128496,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.2,
        "damageTaken": 63221234,
        "deaths": 4,
        "duration": 113314,
        "guild": Object {
          "faction": 1,
          "id": 28064,
          "name": "Slack",
        },
        "healers": 2,
        "melee": 5,
        "ranged": 11,
        "report": Object {
          "code": "GBMzj7md3Z2ALFtf",
          "fightID": 46,
          "startTime": 1599058650697,
        },
        "server": Object {
          "id": 335,
          "name": "Stormreaver",
          "region": "EU",
        },
        "startTime": 1599065817725,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.45,
        "damageTaken": 69866417,
        "deaths": 8,
        "duration": 114533,
        "guild": Object {
          "faction": 1,
          "id": 195039,
          "name": "Legends of Hell",
        },
        "healers": 1,
        "melee": 5,
        "ranged": 12,
        "report": Object {
          "code": "a9MyGj4WqbnTYZgv",
          "fightID": 29,
          "startTime": 1600880952542,
        },
        "server": Object {
          "id": 408,
          "name": "Blackrock",
          "region": "EU",
        },
        "startTime": 1600887312996,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.85,
        "damageTaken": 65207254,
        "deaths": 0,
        "duration": 114653,
        "guild": Object {
          "faction": 1,
          "id": 10239,
          "name": "Vex Thal",
        },
        "healers": 2,
        "melee": 5,
        "ranged": 11,
        "report": Object {
          "code": "2wVxLGWQpXCn8qYy",
          "fightID": 16,
          "startTime": 1597254707642,
        },
        "server": Object {
          "id": 336,
          "name": "Stormscale",
          "region": "EU",
        },
        "startTime": 1597256514282,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.6,
        "damageTaken": 75958436,
        "deaths": 4,
        "duration": 115296,
        "guild": Object {
          "faction": 0,
          "id": 278849,
          "name": "Early Night",
        },
        "healers": 2,
        "melee": 4,
        "ranged": 12,
        "report": Object {
          "code": "mxvkAbzHr2wVWC3t",
          "fightID": 29,
          "startTime": 1595413045972,
        },
        "server": Object {
          "id": 99,
          "name": "Frostmourne",
          "region": "US",
        },
        "startTime": 1595418143712,
        "tanks": 2,
      },
      Object {
        "bracketData": 478.95,
        "damageTaken": 63051900,
        "deaths": 1,
        "duration": 115361,
        "guild": Object {
          "faction": 1,
          "id": 432379,
          "name": "Origin",
        },
        "healers": 1,
        "melee": 8,
        "ranged": 9,
        "report": Object {
          "code": "PJFAXVbtvB7C6zLn",
          "fightID": 7,
          "startTime": 1599504998633,
        },
        "server": Object {
          "id": 384,
          "name": "Sargeras",
          "region": "EU",
        },
        "startTime": 1599506541353,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.35,
        "damageTaken": 70641973,
        "deaths": 1,
        "duration": 116393,
        "guild": Object {
          "faction": 1,
          "id": 11917,
          "name": "傲  血  荣  耀",
        },
        "healers": 2,
        "melee": 5,
        "ranged": 11,
        "report": Object {
          "code": "rwKdpBQnc62ZqCyA",
          "fightID": 5,
          "startTime": 1597321198448,
        },
        "server": Object {
          "id": 676,
          "name": "影之哀伤",
          "region": "CN",
        },
        "startTime": 1597323156136,
        "tanks": 2,
      },
      Object {
        "bracketData": 479.45,
        "damageTaken": 65491807,
        "deaths": 2,
        "duration": 116679,
        "guild": Object {
          "faction": 0,
          "id": 452113,
          "name": "Ethical",
        },
        "healers": 3,
        "melee": 5,
        "ranged": 10,
        "report": Object {
          "code": "PHjdTrc3gpbDwvMG",
          "fightID": 13,
          "startTime": 1597224493534,
        },
        "server": Object {
          "id": 99,
          "name": "Frostmourne",
          "region": "US",
        },
        "startTime": 1597226021024,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.05,
        "damageTaken": 64885019,
        "deaths": 8,
        "duration": 117703,
        "guild": Object {
          "faction": 1,
          "id": 313476,
          "name": "Aurora RF1",
        },
        "healers": 1,
        "melee": 5,
        "ranged": 12,
        "report": Object {
          "code": "RJbK2fXNC3wVM468",
          "fightID": 25,
          "startTime": 1597844330793,
        },
        "server": Object {
          "id": 575,
          "name": "克尔苏加德",
          "region": "CN",
        },
        "startTime": 1597935220932,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.3,
        "damageTaken": 62789294,
        "deaths": 0,
        "duration": 118349,
        "guild": Object {
          "faction": 1,
          "id": 74255,
          "name": "Blur",
        },
        "healers": 2,
        "melee": 5,
        "ranged": 11,
        "report": Object {
          "code": "bmtGxrL2aYZBN9cn",
          "fightID": 1,
          "startTime": 1599619275816,
        },
        "server": Object {
          "id": 222,
          "name": "Tichondrius",
          "region": "US",
        },
        "startTime": 1599619317148,
        "tanks": 2,
      },
      Object {
        "bracketData": 478.8,
        "damageTaken": 72650401,
        "deaths": 6,
        "duration": 118399,
        "guild": Object {
          "faction": 0,
          "id": 76908,
          "name": "The Great Adventure",
        },
        "healers": 1,
        "melee": 7,
        "ranged": 10,
        "report": Object {
          "code": "Kcvxg4DXtT61rmGV",
          "fightID": 13,
          "startTime": 1597795334136,
        },
        "server": Object {
          "id": 57,
          "name": "Dalaran",
          "region": "US",
        },
        "startTime": 1597805700213,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.15,
        "damageTaken": 72202052,
        "deaths": 3,
        "duration": 118603,
        "guild": Object {
          "faction": 0,
          "id": 3929,
          "name": "Vision",
        },
        "healers": 2,
        "melee": 7,
        "ranged": 9,
        "report": Object {
          "code": "bYBtgLT8vKy1j9zG",
          "fightID": 25,
          "startTime": 1597173805223,
        },
        "server": Object {
          "id": 219,
          "name": "Thrall",
          "region": "US",
        },
        "startTime": 1597187897263,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.45,
        "damageTaken": 59133836,
        "deaths": 1,
        "duration": 119097,
        "guild": Object {
          "faction": 1,
          "id": 563170,
          "name": "Oblivion",
        },
        "healers": 2,
        "melee": 4,
        "ranged": 12,
        "report": Object {
          "code": "B87CVQDngtdvWMAG",
          "fightID": 18,
          "startTime": 1600361500824,
        },
        "server": Object {
          "id": 408,
          "name": "Blackrock",
          "region": "EU",
        },
        "startTime": 1600364358696,
        "tanks": 2,
      },
      Object {
        "bracketData": 480.15,
        "damageTaken": 57537737,
        "deaths": 1,
        "duration": 119143,
        "guild": Object {
          "faction": 1,
          "id": 476272,
          "name": "stay hydrated",
        },
        "healers": 2,
        "melee": 6,
        "ranged": 10,
        "report": Object {
          "code": "an9JFfchGjrWAw6M",
          "fightID": 6,
          "startTime": 1600279540372,
        },
        "server": Object {
          "id": 469,
          "name": "Thrall",
          "region": "EU",
        },
        "startTime": 1600281182175,
        "tanks": 2,
      },
      Object {
        "bracketData": 480,
        "damageTaken": 67016397,
        "deaths": 0,
        "duration": 119150,
        "guild": Object {
          "faction": 1,
          "id": 363022,
          "name": "永恒归宿WCL",
        },
        "healers": 3,
        "melee": 5,
        "ranged": 10,
        "report": Object {
          "code": "NCbAn9m2f7BaFckp",
          "fightID": 6,
          "startTime": 1598961513701,
        },
        "server": Object {
          "id": 837,
          "name": "达尔坎",
          "region": "CN",
        },
        "startTime": 1598964502113,
        "tanks": 2,
      },
    ],
  },
  "id": 2333,
  "name": "The Hivemind",
}
`;
